server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/CassandraBlobExistenceTesterIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().delete(defaultBucketName, blobIdDefault).block();
+
+        assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+            .isTrue();
+    }
+
+    @Test
+    void deleteShouldNotDeleteSameBlobIdsFromOtherBucketsInBlobExistenceTester() {
+        BucketName defaultBucketName = testee.getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, SHORT_BYTEARRAY).block();
+
+        testee().delete(defaultBucketName, blobIdDefault).block();
+
+        assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+            .isTrue();
+    }
+
+    @Test
+    void deleteBucketShouldDeleteAllBlobIdsOfABucketInBlobExistenceTester() {
+        BucketName defaultBucketName = testee.getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(defaultBucketName, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().deleteBucket(defaultBucketName).block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(defaultBucketName), blobIdDefault).block())
+                .isFalse();
+            softly.assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(defaultBucketName), blobIdCustom).block())
+                .isFalse();
+        });
+    }
+
+    @Test
+    void deleteBucketShouldDeleteAllBlobIdsWithDifferentBucketsInBlobExistenceTester() {
+        BucketName defaultBucketName = testee.getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().deleteBucket(defaultBucketName).block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(defaultBucketName), blobIdDefault).block())
+                .isFalse();
+            softly.assertThat(cassandraBlobExistenceTester.exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+                .isFalse();
+        });
+    }
+
+    @RepeatedTest(100)
+    void deleteShouldNotAffectSubsequentSaveConsistency() throws Exception {
+        BucketName defaultBucketName = testee.getDefaultBucketName();
+        BlobId blobId = blobIdFactory().forPayload(SHORT_BYTEARRAY);
+
+        ConcurrentTestRunner.builder()
+            .reactorOperation(((threadNumber, step) -> {
+                if (step % 2 == 0) {
```

Comment:
As requested by @rouazana and @mbaechler we can enhance ConcurrentTestRunner to remove some wizardry happening here...